va/validation-authority_test.go

```diff
@@ -90,7 +90,37 @@ func simpleSrv(t *testing.T, token string, stopChan, waitChan chan bool) {
 	}()
 
 	waitChan <- true
-	httpsServer.Serve(conn)
+	if enableTLS {
+		template := &x509.Certificate{
+			SerialNumber: big.NewInt(1337),
+			Subject: pkix.Name{
+				Organization: []string{"tests"},
+			},
+			NotBefore: time.Now(),
+			NotAfter:  time.Now().AddDate(0, 0, 1),
+
+			KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+			ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+			BasicConstraintsValid: true,
+
+			DNSNames: []string{"example.com"},
+		}
+
+		certBytes, _ := x509.CreateCertificate(rand.Reader, template, template, &TheKey.PublicKey, &TheKey)
+		cert := &tls.Certificate{
+			Certificate: [][]byte{certBytes},
+			PrivateKey:  &TheKey,
+		}
+
+		tlsConfig := &tls.Config{
+			Certificates: []tls.Certificate{*cert},
+		}
+
+		tlsListener := tls.NewListener(conn, tlsConfig)
+		server.Serve(tlsListener)
```

Comment:
To avoid a race condition here it might be a good idea to dedup the `server.Serve(*)` calls and move one to after the `if/else` tree as well as moving the `waitChan <- true` statement just before it (but after the logic tree).

Since both `net.Listen` and `tls.NewListener` return `net.Listener` you could add a `var listener net.Listener` statement above the `if/else` tree and assign to it inside it.